galeracluster/source/sslsst.rst

```diff
@@ -0,0 +1,150 @@
+
+   .. code-block:: mysql
+
+      CREATE USER 'sst_user'$'%' IDENTIFIED BY PASSWORD 'sst_password';
+
+   Bear in mind that, due to the manner in which the SST script is called, the user name and password must be the same on all nodes.
+
+#. Grant privileges to this user and require SSL.
+
+   .. code-block:: mysql
+
+      GRANT ALL ON *.* TO 'sst_user'@'%' REQUIRE SSL;
+
+#. Stop the node.  For servers that use ``init``, run the following command:
+
+   .. code-block:: console
+
+      # service mysql stop
+
+   For servers that use ``systemd``, instead run this command:
+
+   .. code-block:: console
+
+      # systemctl stop mysql
+
+#. Using your preferred text editor, ensure that the ``my.cnf`` configuration file includes parameters to secure the database server and client, (``mysqldump`` accesses the server through the client). 
+
+   .. code-block:: ini
+
+      # MySQL Server
+      [mysqld]
+      ssl-ca = /path/to/ca-cert.pem
+      ssl-key = /path/to/server-key.pem
+      ssl-cert = /path/to/server-cert.pem
+
+      # MySQL Client Configuration
+      [client]
+      ssl-ca = /path/to/ca-cert.pem
+      ssl-key = /path/to/client-key.pem
+      ssl-cert = /path/to/client-cert.pem
+
+#. Additionally, configure :ref:`wsrep_sst_auth <wsrep_sst_auth>` with the SST user authentication information.
+
+   .. code-block:: ini
+
+      [mysqld]
+      # mysqldump SST auth
+      wsrep_sst_auth = sst_user:sst_password
+
+This configures the node to use ``mysqldump`` for state snapshot transfers over SSL.  You must repeat this process on all other nodes in order for them to communicate securely with one another.  When all nodes are updated to SSL, you can begin restarting the cluster.  For more information on how to do this, see :doc:`startingcluster`.
+
+
+----------------------------------
+Enabling SSL for ``rsync``
+----------------------------------
+.. _`ssl-rsync`:
+
+The :term:`Physical State Transfer Method` for state snapshot transfers, uses an external script to copy the physical data directly from the file system on one cluster node into another.  In the case of ``rsync``, this method bypasses the database server and client, meaning that you must use an external method to secure its communications through SSL, namely: STunnel.
+
+In order to secure ``rsync`` communications, you need to create a separate SSL certificate for STunnel.  For information on how to generate an SSL certificate, see :doc:`sslcert`.
+
+- ``rsync-ca.pem`` The Certificate Authority (CA) file.
+- ``rsync-cert.pem`` The certificate file.
+- ``rsync-key.pm`` The key file.
+
+Using your preferred text editor, update the Stunnel configuration file at ``/etc/stunnel/stunnel.conf`` with the SSL certificate files for the node.
+
+.. code-block:: ini
+
+   ;; STunnel Configuration
+   CAfile = /path/to/rsync-ca.pem
+   cert = /path/to/rsync-cert.pem
+   key = /path/to/rysnc-key.pem
+
+   ;; ssync Server Configuration
```

Comment:
Believe it was in the original source material.  Not clear on whether or not it's necessary.